lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -136,22 +145,59 @@ class TraceElements extends Gatherer {
 
   /**
    * Find the node ids of elements which are animated using the Animation trace events.
+   * @param {LH.Gatherer.PassContext} passContext
    * @param {Array<LH.TraceEvent>} mainThreadEvents
-   * @return {Array<TraceElementData>}
+   * @return {Promise<Array<TraceElementData>>}
    */
-  static getAnimatedElements(mainThreadEvents) {
-    const animatedElementIds = new Set(mainThreadEvents
-      .filter(e => e.name === 'Animation' && e.ph === 'b')
-      .map(e => this.getNodeIDFromTraceEvent(e)));
+  static async getAnimatedElements(passContext, mainThreadEvents) {
+    const driver = passContext.driver;
+
+    /** @type Map<number, string[]> */
+    const elementAnimations = new Map();
+    mainThreadEvents.filter(e => e.name === 'Animation' && e.ph === 'b')
+      .map(e => {
+        return {
+          nodeId: this.getNodeIDFromTraceEvent(e),
+          animationId: this.getAnimationIDFromTraceEvent(e),
+        };
+      })
+      .forEach(({nodeId, animationId}) => {
+        if (!nodeId || !animationId) return;
+        const animationIds = elementAnimations.get(nodeId);
```

Comment:
just another tool for your toolbox, a pattern you'll frequently find throughout LH is something like

```js
const animationIds = elementAnimations.get(nodeId) || [];
animationIds.push(id);
elementAnimations.set(nodeId, animationIds);
```

but this works fine too :)